Detectors/FIT/common/simulation/src/Digitizer.cxx

```diff
@@ -18,26 +18,83 @@
 #include <algorithm>
 #include <cassert>
 #include <iostream>
+#include <optional>
 
 using namespace o2::fit;
 //using o2::fit::Geometry;
 
 ClassImp(Digitizer);
 
+double sinc(const double x)
+{
+  if (x == 0)
+    return 1;
+  return sin(x) / x;
+}
+double signalForm_i(double x)
+{
+  return -(exp(-0.83344945 * x) - exp(-0.45458 * x)) * (x >= 0) / 7.8446501; // Maximum should be 7.0/250 mV
+};
+
 double get_time(const std::vector<double>& times, double signal_width)
 {
   TH1F hist("time_histogram", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
-  auto signalForm = [](double x) {
-    return -(exp(-0.83344945 * x) - exp(-0.45458 * x));
-  };
-  for (auto time : times)
+  TH1F histsum("time_sum", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  TH1F histshift("time_shift", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  /// Fill Histrogram `hist` with photoelectron induced voltage
+  for (auto time : times) {
     for (int bin = hist.FindBin(time); bin < hist.GetSize(); ++bin)
       if (hist.GetBinCenter(bin) > time)
-        hist.AddBinContent(bin, signalForm(hist.GetBinCenter(bin) - time));
-  double maximum{ hist.GetMaximum() };
-  int binfound{ hist.FindFirstBinAbove(maximum * 0.4) };
-  //  std::cout<<"@@@@@!!! max "<<maximum<<" binfound "<<binfound<<" return "<<hist.GetBinCenter(binfound)<<std::endl;
-  return hist.GetBinCenter(binfound);
+        hist.AddBinContent(bin, signalForm_i(hist.GetBinCenter(bin) - time));
+  }
```

Comment:
As far as I understand, here you fill a cumulative histo for provided times (every value ``time`` is filled many bins). It will be much faster to first fill a simple histo of ``time`` entries, then cumulate them as 
````
double sum=0;
for (int i=0;i<hist.GetSize();i++) {
  sum += hist.GetBinContent(i);
  hist.SetBinContent(i, sum);
}
````

Reply:
the point is - each "time" is photoelectron on the photocathode. Passing MCP signal  enhances and and induces signal with shape  A(t) = -(exp(-0.83344945 * x) - exp(-0.45458 * x)) ; Signal at CFD entrance is sum of these shapes (250 per MIP) plus low frequency noise. CFD=5*A(t-d)-A(t), where d is distance between 0.2 amplitude and maximum. Registered time is 1st point transition between minus and plus.This is why I need all 3 histograms: A(t), A(t-d) and sum. 
I will try your ideas, thank you

Reply:
To move histogram creation to Digitizer is very right. 
Let postponed merge to 1 week

Reply:
Hello
i satisfied your requests. Is it possible possible to merge my branch?

Best regards
Alla

Reply:
I think I fixed everything you mentioned to speed up digitization. It is slow due to huge number of photoelectrons.

Reply:
Thanks! I understand this, so more reasons to speed-up whatever is possible...